java/ql/test/stubs/esapi-2.0.1/org/owasp/esapi/reference/DefaultEncoder.java

```diff
@@ -0,0 +1,8 @@
+package org.owasp.esapi.reference;
+
+import org.owasp.esapi.Encoder;
+
+public class DefaultEncoder implements Encoder {
+  public static Encoder getInstance() { return null; }
+  public String encodeForLDAP(String input) { return input; }
```

Comment:
I think that the reason for one false positive is the fact that the unchanged `input` is being returned here.
Normally, codeql doesn't have access to `DefaultEncoder` source code while building the DB. In this case it has and it knows that `encodeForLDAP` method simply returns unchanged `input`. Therefore it propagates the taint:
```java
// Normally ret isn't tainted, but if codeql has access to encodeForLDAP source, it propagates the taint, so ret is tainted.
String ret = DefaultEncoder.getInstance().encodeForLDAP(tainted);
```
I created the same `DefaultEncoder` stub in my test environment and the false positive popped up. Changing the `encodeForLDAP` method to return `null` fixed it.